test/application_system_test_case.rb

```diff
@@ -19,5 +19,9 @@ class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
 
   # if in CI, run system tests headlessly.
   browser = ENV['GITHUB_WORKFLOW'] ? :headless_chrome : :chrome
+
+  # if in docker, run headless firefox
+  browser = ENV['DOCKER'] ? :headless_firefox : browser
```

Comment:
another way to do this (that involves a little less chaos) is sticking this in a conditional, a la:

```
browser = if ENV['GITHUB_WORKFLOW']
                   :headless_chrome
                 elif ENV['DOCKER']
                   ...
```

that said I'm opposite of worried about this, and merging as is!